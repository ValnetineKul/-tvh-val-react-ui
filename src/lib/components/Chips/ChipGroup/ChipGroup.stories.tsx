import type { ComponentProps } from 'react';
import React from 'react';
import type { StoryTemplate } from '../../../story-utils';
import { createMeta } from '../../../story-utils';
import ChoiceChip from '../ChoiceChip';
import ChipGroup from './ChipGroup';

type Props = ComponentProps<typeof ChipGroup>;
type Args = Omit<Props, 'children'>;

const labels = ['Total Source', 'Deutz', 'CAM', 'Baldwin', 'Danfoss'];

const Template = ((args: Args) => (
  <ChipGroup {...args}>
    {labels.map((label, index) => {
      return <ChoiceChip label={label} key={index} />;
    })}
  </ChipGroup>
)) as StoryTemplate<Args>;

export const Primitive = Template.bind({});
Primitive.args = {
  fillContent: false,
};

Primitive.storyName = 'ChipGroup';

export default createMeta({
  component: ChipGroup,
  title: 'Components/Chips/ChipGroup',
});
