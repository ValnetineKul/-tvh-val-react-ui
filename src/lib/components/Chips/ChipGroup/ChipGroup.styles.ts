import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles({ name: 'ChipGroup' })((theme) => ({
  root: {
    listStyle: 'none',
    marginTop: theme.layoutSpacing['spacing/000'],
    marginBottom: theme.layoutSpacing['spacing/000'],
    paddingLeft: theme.layoutSpacing['spacing/000'],
    display: 'inline-flex',
    flexWrap: 'wrap',
    gap: theme.layoutSpacing['spacing/300'],
  },

  fillContent: {
    display: 'grid',
    gridTemplateColumns: 'repeat(auto-fit, minmax(100px, 1fr))',
    gridAutoRows: 'auto',
    gridGap: theme.layoutSpacing['spacing/300'],
  },
}));

export default useStyles;
