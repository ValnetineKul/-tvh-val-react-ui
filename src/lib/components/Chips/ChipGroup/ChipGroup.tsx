import type { FC, ReactNode } from 'react';
import React from 'react';

import useStyles from './ChipGroup.styles';

export interface ChipGroupProps {
  children: ReactNode;
  fillContent?: boolean;
  className?: string;
}

const ChipGroup: FC<ChipGroupProps> = ({ children, fillContent = false, className }) => {
  const { classes, cx } = useStyles();

  const rootClassName = cx(classes.root, className, {
    [classes.fillContent]: fillContent,
  });

  return (
    <ul className={rootClassName}>
      {children &&
        React.Children.map(children, (child, key) => {
          if (fillContent) {
            return <li key={key}>{React.cloneElement(child as React.ReactElement, { fullWidth: true })}</li>;
          }
          return <li key={key}>{child}</li>;
        })}
    </ul>
  );
};

export default ChipGroup;
