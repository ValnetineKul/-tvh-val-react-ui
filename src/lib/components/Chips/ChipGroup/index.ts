import ChipGroup from './ChipGroup';

export type { ChipGroupProps } from './ChipGroup';
export default ChipGroup;
