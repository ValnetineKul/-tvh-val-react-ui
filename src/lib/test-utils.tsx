import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
