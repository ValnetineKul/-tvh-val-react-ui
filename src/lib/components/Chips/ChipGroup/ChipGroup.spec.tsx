import React from 'react';
import { render } from '../../../test-utils';

import ChipGroup from './ChipGroup';

describe('ChipGroup', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <ChipGroup>
        <div>Item 1</div>
        <div>Item 2</div>
        <div>Item 3</div>
      </ChipGroup>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
