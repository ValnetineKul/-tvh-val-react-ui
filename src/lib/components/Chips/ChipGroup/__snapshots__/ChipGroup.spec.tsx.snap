// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipGroup Should render correctly 1`] = `
<ul
  class="tss-1jia6s6-ChipGroup-root"
>
  <li>
    <div>
      Item 1
    </div>
  </li>
  <li>
    <div>
      Item 2
    </div>
  </li>
  <li>
    <div>
      Item 3
    </div>
  </li>
</ul>
`;
